vector-im/element-android
# Toggle IP address visibility (PSG-860)

## Type of change


## Content
In new Session Management screens we should hide the IP addresses of devices by default for the sake of privacy (in case users may demo these screens to someone else etc.). In this PR, we will provide a menu item to be able to toggle the  visibility of IP addresses. Once a user selects to show or hide the IP addresses then this option is saved to the shared preferences and will be reflected to the screens on next app launches.


## Motivation and context



## Screenshots / GIFs





https://user-images.githubusercontent.com/1254401/200617899-397b271d-bb3e-43d3-9b6c-cfbfc50a9f50.mp4



## Tests



- Enable new session management screen from Labs settings
- Navigate to Settings > Security & Privacy > Show All sessions
- Scroll down to Other Sessions section, click to the menu
- Toggle Show/Hide IP address menu item and see the IP addresses visibility changes
- Click View All button at the bottom of the Other Sessions section
- Toggle Show/Hide IP address menu item and see the changes on UI
- Select a session to navigate to the session overview screen
- Toggle Show/Hide IP address menu item in session overview screen

## Tested devices

- OS version(s):

## Checklist
